Assets/HoloToolkit/UX/Scripts/Lines/LineStripMesh.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using HoloToolkit.Unity;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Holotoolkit.Unity.UX
+{
+    public class LineStripMesh : LineRendererBase
+    {
+        [Header("Strip Mesh Settings")]
+        public Material LineMaterial;
+
+        public Vector3 Forward;
+
+        [SerializeField]
+        private float uvOffset = 0f;
+
+        private Mesh stripMesh;
+        private MeshRenderer stripMeshRenderer;
+        private Material lineMatInstance;
+        private List<Vector3> positions = new List<Vector3>();
+        private List<Vector3> forwards = new List<Vector3>();
+        private List<Color> colors = new List<Color>();
+        private List<float> widths = new List<float>();
+
+        private GameObject meshRendererGameObject;
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+
+            if (LineMaterial == null)
+            {
+                Debug.LogError("Line material cannot be null.");
+                enabled = false;
+                return;
+            }
+
+            lineMatInstance = new Material(LineMaterial);
+
+            // Create a mesh
+            if (stripMesh == null)
+            {
+                stripMesh = new Mesh();
+            }
+
+            if (stripMeshRenderer == null)
+            {
+                meshRendererGameObject = new GameObject("Strip Mesh Renderer");
+                stripMeshRenderer = meshRendererGameObject.AddComponent<MeshRenderer>();
+            }
+            stripMeshRenderer.sharedMaterial = lineMatInstance;
+            stripMeshRenderer.shadowCastingMode = UnityEngine.Rendering.ShadowCastingMode.Off;
+            stripMeshRenderer.receiveShadows = false;
+            stripMeshRenderer.lightProbeUsage = UnityEngine.Rendering.LightProbeUsage.Off;
+
+            MeshFilter stripMeshFilter = stripMeshRenderer.GetComponent<MeshFilter>();
+            if (stripMeshFilter == null)
+                stripMeshFilter = stripMeshRenderer.gameObject.AddComponent<MeshFilter>();
```

Comment:
we could use `gameObject.EnsureComponent<MeshFilter>()` here too.
Also : braces

Reply:
Good idea. Really useful extension.